stable/fluentd/templates/deployment.yaml

```diff
@@ -12,17 +12,20 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
 spec:
+{{- if .Values.autoscaling.enabled}}
+  serviceName: {{ template "fluentd.fullname" . }}
```

Comment:
StatefulSets require a headless service which this one isn't. You'd have to create an additional headless service.

See https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#limitations

Reply:
Done, please check.